[Bug]: address sanitizer not installed on amazon linux runner

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

/usr/bin/ld: cannot find /usr/lib64/libasan.so.6.0.0: No such file or directory
collect2: error: ld returned 1 exit status

### Expected Behavior

address sanitizer should be installed on amazon linux runner

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

--- a/build/docker/builder/cpu/amazonlinux2023/Dockerfile
+++ b/build/docker/builder/cpu/amazonlinux2023/Dockerfile
@@ -15,7 +15,7 @@ ARG TARGETARCH
 
 RUN dnf install -y wget g++ gcc gdb libatomic libstdc++-static ninja-build git make zip unzip tar which \
     autoconf automake golang python3 python3-pip perl-FindBin texinfo \
-    pkg-config libuuid-devel libaio perl-IPC-Cmd && \
+    pkg-config libuuid-devel libaio perl-IPC-Cmd libasan && \
     rm -rf /var/cache/yum/*
 
 RUN pip3 install conan==1.61.0

